### ISSUE TITLE
Request - reorganize database roles

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#9218]
Help us understand your request (check below):

- [x] other

**Describe what you're trying to do** 

https://github.com/ArctosDB/internal/issues/447 has turned into a proposal for new roles; if we're doing everything by "package" (list of current roles) then there's no need for the complexity of "subroles." There's clearly some confusion around user management, simplifying is a worthy goal.

"Roles" are primarily shortcuts to assigning permissions (Create, Read, Update, Delete) on tables. We can have any number of them, and they can overlap in any way. (Role1 and Role2 can both have UPDATE on **table1**.) The "proper" number is about whatever results in the most intuitive user management without giving anyone access they're not ready for.

Roles also drive form access in Arctos, and can exist solely for that purpose. That is, it's possible to have one role which allows UPDATE on **table1**, and another which provides access to scary forms which update **table1** (eg bulk tools) - and the latter could rely on the former, or carry explicit permissions. (I suspect cascading dependencies rather than overlaps are simpler to understand, but there are no real technical limitations in that.)

I think the big jumps worthy of separation are probably

* access to live data - the current packages couple data entry (write to a temp table which requires review) with other stuff, I think it should not
* access to mint identifiers, which should be tightly coupled to an institutional commitment
* access to bulk tools, which provide the possibility of millions of oopsies with a single click
* access to shared data, which allows oopsies to spread beyond 'your' collections
* "admin stuff" - but I think there's some legacy complexity in that too, and it should also be reviewed if we're doing anything revolutionary

https://docs.google.com/spreadsheets/d/1ZH2YcYXAFpOW-cR2jEXwmxt3hTB1b7A34qHohg4M98E/edit?usp=sharing was built from the current packages and roles, and hopefully serves as a useful starting point.

https://arctos.database.museum/Admin/user_roles.cfm (should be very familiar to many of us!) contains all the same data, and some documentation.

HELP! (And let me know if I can help by shuffling data in some way.)

Ping @mkoo 

------------

Please review the [Arctos Code of Conduct](https://arctosdb.org/code-of-conduct/) to ensure respectful communication, and see <https://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html> for more information about submitting and managing Issues.

**Comment by mkoo** [comment]
Since this is somehow all my fault, so I am diving in and I think your big jumps are the place to launch from first. So I am going to start with a roadmap of these bullet pointed "roles"/permission levels and go from there....

editing in [cf_ctuser_roles ](https://docs.google.com/spreadsheets/d/1ytYgKaJ84HGCb5c2oRs5tAMUUkrrFohYCIBYYExurGA/edit?gid=1064277392#gid=1064277392)

**Comment by dustymc** [comment]
Dumping https://github.com/ArctosDB/arctos/issues/8180 into here; project creation needs handled in some way.

**Comment by dustymc** [comment]
https://github.com/ArctosDB/arctos-roadmap/issues/24 is back to a UI-only task; closing this as wontfix.
